servers/rendering/rendering_device_binds.cpp

```diff
@@ -31,6 +31,7 @@
 #include "rendering_device_binds.h"
 
 Error RDShaderFile::parse_versions_from_text(const String &p_text, const String p_defines, OpenIncludeFunction p_include_func, void *p_include_func_userdata) {
+	ERR_FAIL_COND_V_EDMSG((DisplayServer::get_singleton()->get_name() == "headless"), ERR_UNAVAILABLE, "Cannot import custom .glsl shaders when running in headless mode.");
```

Comment:
This should already fail on the next line. The fact that it isn't highlights that `get_singleton()` is returning a valid pointer when it shouldn't. Therefore the solution is going to be ensuring that the singleton isn't initialized when RD isn't available

Reply:
You are correct that it does indeed fail when get_singleton is null.  I believe my 

```cpp
void BetsyCompressor::_init() {
	if (!DisplayServer::can_create_rendering_device()) {
		return;
	}
```

and

```
bool DisplayServer::can_create_rendering_device() {
	if (get_singleton()->get_name() == "headless") {
		return false;
	}
```

Fixes the issue of the singleton being created.

But then on top of that, I just thought this error message was more helpful than the null error message regarding the singleton.  If you prefer, I can remove the change you commented on and I believe it will still work.  It will just throw "null get_singleton" error messages instead of the "Cannot import custom .glsl shaders when running in headless mode." error messages.  Whichever you prefer is fine with me.

TL; DR;

You are correct.  My other 2 changes fix the issue of that singleton being created.  If you would like, I can remove this change.

Reply:
It is a much better error message, but I would prefer to improve the error message for the old error instead of doing a redundant check. 

Perhaps we could add the following error message to the old check ""Cannot import custom .glsl shaders when running without a RenderingDevice. This can happen if you are using the headless more or the Compatibility backend."

Reply:
done